ntp-proto/src/algorithm/kalman/peer.rs

```diff
@@ -461,7 +466,49 @@ impl PeerState {
 
     pub fn snapshot<Index: Copy>(&self, index: Index) -> Option<PeerSnapshot<Index>> {
         match &self.0 {
-            PeerStateInner::Initial(_) => None,
+            PeerStateInner::Initial(InitialPeerFilter {
+                roundtriptime_stats,
+                init_offset,
+                last_measurement: Some(last_measurement),
+                samples,
+            }) if *samples > 0 => {
+                let max_roundtrip = roundtriptime_stats.data[..*samples as usize]
+                    .iter()
+                    .copied()
+                    .fold(None, |v1, v2| {
+                        if v2.is_nan() {
+                            v1
+                        } else if let Some(v1) = v1 {
+                            if v1 > v2 {
+                                Some(v1)
+                            } else {
+                                Some(v2)
+                            }
```

Comment:
maybe use `Some(Ord::max(v1, v2))`

Reply:
Can't, f64 is not Ord (only partial_ord because of NaN)